File: SEMESTRE 3/C++/EXTRAS/POINTEURS/exo2.cpp
/*Ecrire un programme qui permet de saisir N de type entier.
 -Une fonction qui permet de saisir N 
 -Une fonction qui permet de vérifier si N est premier 
NB : N est utilisé comme pointeur*/

#include <iostream>
#include <cstdlib>

void saisie(int *n){
	
	std::cout << "Saisir n:  ";
	std::cin >> *n;
}

bool estPremier(int* n){
	if(*n < 2) return false;

	for (int i = 2; i < *n; i++)
	{
		if (*n % i == 0)
		{
			return false;
		}
	}
	return true;
}

int main()
{
	int *n = new int;
	saisie(n);
	std::cout << " \n" << *n << std::endl;
	std::cout << "------------------ \n" << std::endl;
	std::cout << estPremier(n);
	
	return 0;
}

File: SEMESTRE 3/PYTHON/Python/COURSES/Chapitre 2/exo2.py
##ecrire une fonction qui prend un entier en argument et
##retourne True si le nombre est premier
def estpremier(nbre):
    cpt = 0
    for i in range(1,nbre+1):
        if nbre % i == 0 :
            cpt +=1
    if cpt==2 :
        return True
    else:
        return False
nbre=int(input("entrer un nombre entier"))
if estpremier(nbre):
    print("le nombre est premier")
else:
    print("le nombre n'est pas premier")

# ecrire un fonction premier qui prend enargument une liste et qui
# retourne une liste contenant la somme des nombre pairs et
# le produit des nombres impairs
